### DISCUSSION TITLE
Difference between array & vector

### DISCUSSION THREAD

**Discussion by mykola-kobets-epam** [ETLCPP/etl#882] [General]
It's not very clear why both implementations: array & vector provided. Even though each one has a static capacity.
I think array & vector_ext would be enough. 
Thank You.

**Comment by jwellbelove** [comment]
`etl::array` has a _fixed_ size and capacity, `etl::vector` & `etl::vector_ext` have _variable_ size and _fixed_ capacity.
`etl::vector` is usually more convenient to use than an `etl::vector_ext`, as it contains its own storage.
`etl::vector_ext` is useful when the storage is require to be placed in a specific location, such as fast static or internal RAM, or allocated from a memory pool.
There are several containers in the ETL that have `_ext` variants.
